### PR TITLE
docs: Changed remaining links to sources to API docs

### PR DESCRIPTION
**Pull request by vlastahajek**
- Change links leading to sources to go to API docs
- Fixed broken links to InfluxDB docs

## Checklist

- [X] CHANGELOG.md updated
- [X] Rebased/mergeable
- [X] Tests pass
- [X] Commit messages are in [semantic format](https://seesparkbox.com/foundry/semantic_commit_messages)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ This repository contains the reference Go client for InfluxDB 2.
     - Writing data using
         - [Line Protocol](https://docs.influxdata.com/influxdb/v1.6/write_protocols/line_protocol_tutorial/) 
         - [Data Point](https://pkg.go.dev/github.com/influxdata/influxdb-client-go/v2@v2.2.0/api/write#Point)
-        - Both [asynchronous](https://github.com/influxdata/influxdb-client-go/blob/master/api/write.go) or [synchronous](https://github.com/influxdata/influxdb-client-go/blob/master/api/writeAPIBlocking.go) ways
+        - Both [asynchronous](https://pkg.go.dev/github.com/influxdata/influxdb-client-go/v2@v2.2.0/api#WriteAPI) or [synchronous](https://pkg.go.dev/github.com/influxdata/influxdb-client-go/v2@v2.2.0/api#WriteAPIBlocking) ways
```
Comment (sranka) [inline README.md:33]:
```suggestion
        - Both [asynchronous](https://pkg.go.dev/github.com/influxdata/influxdb-client-go/v2/api#WriteAPI) or [synchronous](https://pkg.go.dev/github.com/influxdata/influxdb-client-go/v2/api#WriteAPIBlocking) ways
```

File: README.md [hunk 2]
```diff
@@ -40,15 +40,15 @@ This repository contains the reference Go client for InfluxDB 2.
...
 
 ### Examples
 
 Examples for basic writing and querying data are shown below in this document 
 
 There are also other examples in the API docs:
- - [Client usage](https://pkg.go.dev/github.com/influxdata/influxdb-client-go?tab=doc#pkg-examples) 
- - [Management APIs](https://pkg.go.dev/github.com/influxdata/influxdb-client-go@v1.4.0/api?tab=doc#pkg-examples) 
+ - [Client usage](https://pkg.go.dev/github.com/influxdata/influxdb-client-go/v2?tab=doc#pkg-examples) 
+ - [Management APIs](https://pkg.go.dev/github.com/influxdata/influxdb-client-go/v2@v2.2.0/api?tab=doc#pkg-examples)
```
Comment (sranka) [inline README.md:51]:
```suggestion
 - [Management APIs](https://pkg.go.dev/github.com/influxdata/influxdb-client-go/v2/api?tab=doc#pkg-examples) 
```

File: README.md [hunk 3]
```diff
@@ -300,7 +300,7 @@ func main() {
 ```
 
 ### Queries
-Query client offers retrieving of query results to a parsed representation in a `QueryTableResult` or to a raw string. 
+Query client offers retrieving of query results to a parsed representation in a [QueryTableResult](https://pkg.go.dev/github.com/influxdata/influxdb-client-go/v2@v2.2.0/api#QueryTableResult) or to a raw string.
```
Comment (sranka) [inline README.md:303]:
```suggestion
Query client offers retrieving of query results to a parsed representation in a [QueryTableResult](https://pkg.go.dev/github.com/influxdata/influxdb-client-go/v2/api#QueryTableResult) or to a raw string. 
```

File: README.md [hunk 4]
```diff
@@ -392,9 +392,9 @@ func main() {
...
-  | [WriteAPI](write.go) (also [WriteAPIBlocking](writeAPIBlocking.go))| [/api/v2/write](https://docs.influxdata.com/influxdb/latest/tools/api/#api-v2-write-http-endpoint) | Write data to InfluxDB 1.8.0+ using the InfluxDB 2.0 API |
-  | [QueryAPI](query.go) | [/api/v2/query](https://docs.influxdata.com/influxdb/latest/tools/api/#api-v2-query-http-endpoint) | Query data in InfluxDB 1.8.0+ using the InfluxDB 2.0 API and [Flux](https://docs.influxdata.com/flux/latest/) endpoint should be enabled by the [`flux-enabled` option](https://docs.influxdata.com/influxdb/latest/administration/config/#flux-enabled-false)
-  | [Health()](client.go#L55) | [/health](https://docs.influxdata.com/influxdb/latest/tools/api/#health-http-endpoint) | Check the health of your InfluxDB instance |    
+  | [WriteAPI](https://pkg.go.dev/github.com/influxdata/influxdb-client-go/v2@v2.2.0/api#WriteAPI) (also [WriteAPIBlocking](https://pkg.go.dev/github.com/influxdata/influxdb-client-go/v2@v2.2.0/api#WriteAPIBlocking))| [/api/v2/write](https://docs.influxdata.com/influxdb/latest/tools/api/#api-v2-write-http-endpoint) | Write data to InfluxDB 1.8.0+ using the InfluxDB 2.0 API |
```
Comment (sranka) [inline README.md:395]:
```suggestion
  | [WriteAPI](https://pkg.go.dev/github.com/influxdata/influxdb-client-go/v2/api#WriteAPI) (also [WriteAPIBlocking](https://pkg.go.dev/github.com/influxdata/influxdb-client-go/v2/api#WriteAPIBlocking))| [/api/v2/write](https://docs.influxdata.com/influxdb/latest/tools/api/#api-v2-write-http-endpoint) | Write data to InfluxDB 1.8.0+ using the InfluxDB 2.0 API |
```

File: README.md [hunk 5]
```diff
@@ -392,9 +392,9 @@ func main() {
...
+  | [QueryAPI](https://pkg.go.dev/github.com/influxdata/influxdb-client-go/v2@v2.2.0/api#QueryAPI) | [/api/v2/query](https://docs.influxdata.com/influxdb/latest/tools/api/#api-v2-query-http-endpoint) | Query data in InfluxDB 1.8.0+ using the InfluxDB 2.0 API and [Flux](https://docs.influxdata.com/flux/latest/) endpoint should be enabled by the [`flux-enabled` option](https://docs.influxdata.com/influxdb/latest/administration/config/#flux-enabled-false)
```
Comment (sranka) [inline README.md:396]:
```suggestion
  | [QueryAPI](https://pkg.go.dev/github.com/influxdata/influxdb-client-go/v2/api#QueryAPI) | [/api/v2/query](https://docs.influxdata.com/influxdb/latest/tools/api/#api-v2-query-http-endpoint) | Query data in InfluxDB 1.8.0+ using the InfluxDB 2.0 API and [Flux](https://docs.influxdata.com/flux/latest/) endpoint should be enabled by the [`flux-enabled` option](https://docs.influxdata.com/influxdb/latest/administration/config/#flux-enabled-false)
```

File: README.md [hunk 6]
```diff
@@ -30,7 +30,7 @@ This repository contains the reference Go client for InfluxDB 2.
     - Writing data using
         - [Line Protocol](https://docs.influxdata.com/influxdb/v1.6/write_protocols/line_protocol_tutorial/) 
         - [Data Point](https://pkg.go.dev/github.com/influxdata/influxdb-client-go/v2@v2.2.0/api/write#Point)
```
Comment (sranka) [inline README.md:32]:
```suggestion
        - [Data Point](https://pkg.go.dev/github.com/influxdata/influxdb-client-go/v2/api/write#Point)
```

**Review by sranka — CHANGES_REQUESTED** [review]
Look good, but I think that we should not reference an exact version (v2.2.0) in markdown files, but rather the latest version of v2.

**Comment by vlastahajek** [comment]
@sranka, thanks for the suggestions. I wanted to have links without a version, but it didn't work for me, for some reason. 
But it is OK now.

**Review by ivankudibal — APPROVED** [review]

File: README.md (hunk 6 above)
Comment (vlastahajek) [reply README.md]: Done

File: README.md (hunk 1 above)
Comment (vlastahajek) [reply README.md]: Done

File: README.md (hunk 2 above)
Comment (vlastahajek) [reply README.md]: Done

File: README.md (hunk 3 above)
Comment (vlastahajek) [reply README.md]: Done

File: README.md (hunk 4 above)
Comment (vlastahajek) [reply README.md]: Done

File: README.md (hunk 5 above)
Comment (vlastahajek) [reply README.md]: Done

**Pull request merged**